spark/dl/src/main/scala/com/intel/analytics/bigdl/example/optimizerWrapper/OptimizerWrapper.scala

```diff
@@ -0,0 +1,127 @@
+        val sqLContext = new SQLContext(sc)
+        Engine.init
+
+        val trainData = Paths.get(param.folder, "/train-images-idx3-ubyte")
+        val trainLabel = Paths.get(param.folder, "/train-labels-idx1-ubyte")
+        val validationData = Paths.get(param.folder, "/t10k-images-idx3-ubyte")
+        val validationLabel = Paths.get(param.folder, "/t10k-labels-idx1-ubyte")
+
+        val model = if (param.modelSnapshot.isDefined) {
+          Module.load[Float](param.modelSnapshot.get)
+        } else {
+          LeNet5(classNum = 10)
+        }
+
+        val state = if (param.stateSnapshot.isDefined) {
+          T.load(param.stateSnapshot.get)
+        } else {
+          T(
+            "learningRate" -> param.learningRate,
+            "learningRateDecay" -> param.learningRateDecay
+          )
+        }
+
+        val trainSet = DataSet.array(load(trainData, trainLabel), sc) ->
+          BytesToGreyImg(28, 28) -> GreyImgNormalizer(trainMean, trainStd) -> GreyImgToBatch(
+          param.batchSize)
+
+        val validationSet = DataSet.array(load(validationData, validationLabel), sc) ->
+          BytesToGreyImg(28, 28) -> GreyImgNormalizer(testMean, testStd) -> GreyImgToBatch(
+          param.batchSize)
+
+        val dataFrameRDD : RDD[DLEstimatorData[Float]] = trainSet.
+          asInstanceOf[DistributedDataSet[MiniBatch[Float]]].data(false).map(batch => {
+            val feature = batch.data
+            val label = batch.labels
+          val estimatorData = DLEstimatorData(DLEstimatorMinibatchData(feature.storage().array(),
+            feature.size(), label.storage().array(), label.size()))
+            estimatorData
+        })
+
+        var df : DataFrame = sqLContext.createDataFrame(dataFrameRDD).toDF("minibatch")
+
+        val criterion = ClassNLLCriterion[Float]()
+
+        val estimator = new DLEstimator[Float](model, criterion,
+          Array[Int](128, 28, 28))("DLEstimator")
+
+        val transformer = estimator.fit(df)
+
+        val rdd: RDD[DenseVectorData] = validationSet.
+          asInstanceOf[DistributedDataSet[MiniBatch[Float]]].data(false).flatMap(batch => {
+          val buffer = new ArrayBuffer[DenseVectorData]()
+          val feature = batch.data.storage().toArray
+          var i = 0
+          while (i < 128) {
+            val next = new DenseVector(feature.
+              slice(i * 28 * 28, (i + 1) * 28 * 28).map(_.toDouble))
+            val data = DenseVectorData(next)
+            buffer.append(data)
+            i += 1
+          }
+          buffer.iterator
+        })
+        var transDF : DataFrame = sqLContext.createDataFrame(rdd).toDF("features")
+        val transformed = transformer.transform(transDF)
+        transformed.select("features", "predict")
+          .collect()
+          .foreach { case Row(data: DenseVector, predict: Int) =>
+            println(data + "=>" + predict)
+          }
+
+      })
+    }
+}
+case class DenseVectorData(denseVector : DenseVector)
```

Comment:
Dataframe  recursively parse the struct type, this case class is just to ensure the input feature is in one column